source/common/network/happy_eyeballs_connection_impl.cc

```diff
@@ -379,6 +380,41 @@ void HappyEyeballsConnectionImpl::dumpState(std::ostream& os, int indent_level)
   }
 }
 
+std::vector<Address::InstanceConstSharedPtr>
+HappyEyeballsConnectionImpl::sortAddresses(const std::vector<Address::InstanceConstSharedPtr>& in) {
+  std::vector<Address::InstanceConstSharedPtr> address_list = in;
+  for (size_t current = 1; current < address_list.size(); ++current) {
+    // If the current address has a different family than the previous address then
+    // it is in the correct position.
+    if (address_list[current]->type() != Address::Type::Ip ||
+        address_list[current - 1]->type() != Address::Type::Ip ||
+        address_list[current]->ip()->version() != address_list[current - 1]->ip()->version()) {
+      continue;
+    }
+    // Find the first address with a different address family than the current address.
+    size_t next = current + 1;
+    while (next < address_list.size()) {
```

Comment:
nit: this could be implemented with [find_if](https://www.cplusplus.com/reference/algorithm/find_if/)